src/gleam_stdlib.erl

```diff
@@ -498,6 +498,13 @@ inspect_maybe_utf8_string(Binary, Acc) ->
                 $\n -> <<$\\, $n>>;
                 $\t -> <<$\\, $t>>;
                 $\f -> <<$\\, $f>>;
+                127 -> <<$\\, $u, ${, $0, $0, $7, $F, $}>>;
+                X when X < 32 ->
+                    Hex = integer_to_list(X, 16),
+                    Leading = lists:duplicate(4 - length(Hex), "0"),
+                    Formatted = lists:append(Leading, Hex),
+                    Bin = list_to_binary(Formatted),
+                    <<$\\, $u, ${, Bin/binary, $}>>;
```

Comment:
You could consider simplifying the above lines to something like this: `list_to_binary(io_lib:format("\\u{~4.16.0B}", [X]))` --it's a bit cryptic, but the `io_lib:format` is used elsewhere in the module at least.

(That format is `~4.16.0B` saying, width=4, base 16 int, left pad with `0`.)

Reply:
I'm inexperienced with Erlang so I didn't even realize `io_lib:format` was a thing. I'll look into it!